tests/integration/docusaurus/validation/checkpoints/how_to_pass_an_in_memory_dataframe_to_a_checkpoint.py

```diff
@@ -40,9 +38,8 @@
 
 batch_request = dataframe_asset.build_batch_request()
 
-checkpoint = SimpleCheckpoint(
+checkpoint = context.add_or_update_checkpoint(
```

Comment:
@donaldheppner I cannot see this in the pull request, but I have an important question: after creating a `Checkpoint` with a `DataFrame` DataAsset, do we also run this `Checkpoint` as part of this documentation item's integration test?  If not, then we must do so, because I expect the call to `run()` to raise an exception.  If that happens, then the correct `run()` method would be: `run_with_runtime_args()` (where we can supply `batch_request` or `validator`)).  Happy to discuss.  Thank you.

Reply:
yes, the checkpoint is run as part of the test

Reply:
@donaldheppner I just looked at the file, and it seems to me that the actual (dynamic object) `DataFrame` was not passed to `build_batch_request()`, implying that the `run()` method did not really have a `Batch` to work with.  Prior to running the `Checkpoint`, we need to say `batch_request = dataframe_asset.build_batch_request(dataframe=df)`, where `df` is our `DataFrame` from line 9 (where we read in the "CSV" file).  Only then the behavior of the run method would be indicative of success.  Thank you.

Reply:
`context.sources.add_pandas("taxi_datasource").read_dataframe(
    df, asset_name="taxi_frame", batch_metadata={"year": "2019", "month": "01"}
)` accepts the dataframe, that should be sufficient? I loathe to add extra lines to get this trivial example to work

Reply:
@donaldheppner As far as I can tell, the only way to ensure that the `dataframe` is set on the `DataFrameAsset` is via `batch_request = dataframe_asset.build_batch_request(dataframe=df)`; could we please try it to make sure that running the `Checkpoint` after that will work properly?  Thank you.